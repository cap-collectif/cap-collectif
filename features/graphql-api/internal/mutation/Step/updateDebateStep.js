/* eslint-env jest */
import '../../../_setup';

const UpdateDebateStepMutation = /* GraphQL*/ `
  mutation UpdateDebateStepMutation($input: UpdateDebateStepInput!) {
    updateDebateStep(input: $input) {
            debateStep {
                title
                label
                body
                timeRange {
                    startAt
                    endAt
                }
                enabled
                timeless
                isAnonymousParticipationAllowed
                metaDescription
                customCode
                debateType
                debateContent
                debate {
                    id
                    articles {
                        edges {
                            node {
                              id 
                              url
                            }
                        }
                    }
                }
            }
        }
  }
`

const input = {
  "id": "RGViYXRlU3RlcDpkZWJhdGVTdGVwQ2FubmFiaXM=",
  "label": "Débatzz",
  "body": "body",
  "title": "Pour ou contre la légalisation du Cannabis ?",
  "startAt": "2023-10-01 14:00:00",
  "endAt": "2037-10-01 12:00:00",
  "isEnabled": true,
  "timeless": false,
  "isAnonymousParticipationAllowed": false,
  "metaDescription": "meta description",
  "customCode": "custom code",
  "debateType": "FACE_TO_FACE",
  "debateContent": "content",
  "articles": [
    {
      "id": "RGViYXRlQXJ0aWNsZTpjYW5hYmlzQXJ0aWNsZUJmbQ==",
      "url": "https://www.bfmtv.com/economie/estime-a-3-2-milliards-d-euros-en-france-le-marche-du-cannabis-serait-bien-plus-gros-que-ce-qu-on-pensait_AN-202010200002.html"
    },
    {
      "id": "RGViYXRlQXJ0aWNsZTpjYW5hYmlzQXJ0aWNsZU91ZXN0RnJhbmNlMQ==",
      "url": "https://www.ouest-france.fr/societe/cannabis/cannabis-therapeutique-le-cahier-des-charges-des-futurs-medicaments-publie-7020873"
    }
  ]
}

describe('mutations.updateDebateStep', () => {
  it('admin should be able to edit debate step.', async () => {
    const id = toGlobalId('DebateStep', 'debateStepCannabis'); // RGViYXRlU3RlcDpkZWJhdGVTdGVwQ2FubmFiaXM=
    const response = await graphql(
      UpdateDebateStepMutation,
      {input: {...input, id}},
      'internal_admin',
    );
    expect(response).toMatchSnapshot();
  });
});
