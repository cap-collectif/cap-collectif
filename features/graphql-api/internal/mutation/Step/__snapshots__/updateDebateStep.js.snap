// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations.updateDebateStep admin should be able to edit debate step. 1`] = `
Object {
  "updateDebateStep": Object {
    "debateStep": Object {
      "body": "body",
      "customCode": "custom code",
      "debate": Object {
        "articles": Object {
          "edges": Array [
            Object {
              "node": Object {
                "id": "RGViYXRlQXJ0aWNsZTpjYW5hYmlzQXJ0aWNsZUJmbQ==",
                "url": "https://www.bfmtv.com/economie/estime-a-3-2-milliards-d-euros-en-france-le-marche-du-cannabis-serait-bien-plus-gros-que-ce-qu-on-pensait_AN-202010200002.html",
              },
            },
            Object {
              "node": Object {
                "id": "RGViYXRlQXJ0aWNsZTpjYW5hYmlzQXJ0aWNsZU91ZXN0RnJhbmNlMQ==",
                "url": "https://www.ouest-france.fr/societe/cannabis/cannabis-therapeutique-le-cahier-des-charges-des-futurs-medicaments-publie-7020873",
              },
            },
          ],
        },
        "id": "RGViYXRlOmRlYmF0ZUNhbm5hYmlz",
      },
      "debateContent": "content",
      "debateType": "FACE_TO_FACE",
      "enabled": true,
      "isAnonymousParticipationAllowed": false,
      "label": "Débatzz",
      "metaDescription": "meta description",
      "timeRange": Object {
        "endAt": "2037-10-01 12:00:00",
        "startAt": "2023-10-01 14:00:00",
      },
      "timeless": false,
      "title": "Pour ou contre la légalisation du Cannabis ?",
    },
  },
}
`;
